src/Common/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -88,65 +89,117 @@ public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
             if (padding == null)
                 throw new ArgumentNullException(nameof(padding));
 
-            Interop.Crypto.RsaPadding rsaPadding = GetInteropPadding(padding);
+            Interop.Crypto.RsaPadding rsaPadding = GetInteropPadding(padding, out RsaPaddingProcessor oaepProcessor);
             SafeRsaHandle key = _key.Value;
             CheckInvalidKey(key);
 
-            byte[] buf = new byte[Interop.Crypto.RsaSize(key)];
+            int rsaSize = Interop.Crypto.RsaSize(key);
+            byte[] buf = null;
 
-            int returnValue = Interop.Crypto.RsaPrivateDecrypt(
-                data.Length,
-                data,
-                buf,
-                key,
-                rsaPadding);
+            try
+            {
+                buf = ArrayPool<byte>.Shared.Rent(rsaSize);
+                Span<byte> destination = new Span<byte>(buf, 0, rsaSize);
 
-            CheckReturn(returnValue);
+                if (!TryDecrypt(key, data, destination, rsaPadding, oaepProcessor, out int bytesWritten))
+                {
+                    Debug.Fail($"{nameof(TryDecrypt)} should not return false for RSA_size buffer");
+                    throw new CryptographicException();
+                }
 
-            // If the padding mode is RSA_NO_PADDING then the size of the decrypted block
-            // will be RSA_size, so let's just return buf.
-            //
-            // If any padding was used, then some amount (determined by the padding algorithm)
-            // will have been reduced, and only returnValue bytes were part of the decrypted
-            // body, so copy the decrypted bytes to an appropriately sized array before
-            // returning it.
-            if (returnValue == buf.Length)
+                return destination.Slice(0, bytesWritten).ToArray();
+            }
+            finally
             {
-                return buf;
+                if (buf != null)
+                {
+                    Array.Clear(buf, 0, rsaSize);
```

Comment:
How are you deciding when to use Array.Clear vs CryptographicOperation.ZeroMemory?

Reply:
Honest: The parts I wrote before rebasing are Array.Clear, the parts after are ZeroMemory... unless they're in a file that gets built for something other than netcoreapp.

Less honest: Just the !netcoreapp part, but with a bad self-analysis.

Reply:
Normalized to ZeroMemory